workers/src/main/scala/workers/itc/ITCRequests.scala

```diff
@@ -60,70 +62,41 @@ object ITCRequests:
-                signalToNoise = params.signalToNoise,
-                sourceProfile = params.target.profile.toInput,
-                signalToNoiseAt = params.signalToNoiseAt.map(_.toInput).orIgnore,
-                band = band,
-                radialVelocity = params.target.rv.toITCInput,
-                constraints = params.constraints,
-                modes = params.mode.toITCInput.map(_.assign).toList
-              ).assign
-            )
-            .flatTap { r =>
-              val prefix = s"ITC: Result for mode ${params.mode}:"
-              itcResults(r).traverse(_ match {
-                case Left(errors)                                     =>
-                  Logger[F].error(s"$prefix ERRORS: $errors")
-                case Right(ItcResult.Result(exposureTime, exposures)) =>
-                  Logger[F].debug(s"$prefix $exposures x ${exposureTime.toSeconds}")
-                case Right(other)                                     =>
-                  Logger[F].debug(s"$prefix $other")
-              })
-            }
-            .map(r =>
-              itcResults(r) match {
-                case Nil => none
-                case l   =>
-                  Map(
-                    params ->
-                      l.maxBy {
-                        case Right(ItcResult.Result(exposureTime, _)) => exposureTime.toMicros
-                        case _                                        => Long.MinValue
-                      }
-                  ).some
-              }
-            )
-            .handleErrorWith(e =>
-              Map(
-                params -> (ItcQueryProblems
-                  .GenericError("Error calling ITC service"): ItcQueryProblems)
-                  .leftNec[ItcResult]
-              ).some.pure[F]
-            )
-        }
-        .sequence
-        .map(_.flatten)
+        ) *> selectedBand(params.target.profile, params.wavelength.value).map { band =>
+        SpectroscopyITCQuery[F]
+          .query(
+            SpectroscopyIntegrationTimeInput(
+              wavelength = params.wavelength.value.toInput,
+              signalToNoise = params.signalToNoise,
+              sourceProfile = params.target.profile.toInput,
+              signalToNoiseAt = params.signalToNoiseAt.map(_.toInput).orIgnore,
+              band = band,
+              radialVelocity = params.target.rv.toITCInput,
+              constraints = params.constraints,
+              mode = params.mode.toITCInput.orIgnore
+            ).assign
+          )
+          .map(r => Map(params -> itcResults(r)))
+          .handleError { e =>
+            val msg = e match
+              case ResponseException(errors) =>
+                errors.map(_.message).mkString_("\n")
+              case e                         =>
+                e.getMessage
+            Map(params -> (ItcQueryProblems.GenericError(msg): ItcQueryProblems).leftNec[ItcResult])
+          }
+      }.sequence
```

Comment:
Minor thing, but can't `.map(...).sequence` be replaced by `.traverse`?

Reply:
The answer is always traverse.